Twilio-org/phonebank
# Ticket 123: Set up TwiML handler for starting volunteer calls

***DO NOT MERGE BEFORE #139 ***

`export const lintDummy = { twilioClient, CALLER_ID };` in `/server/util/twilil.js` is to make linting pass. without it, linter complains for `Prefer Default Export` and `No Unused Vars`. This should be removed after #139 gets merged and this work will exist int the same file.

## Manual Testing

1. Do npm run db:refresh to empty your database, recreate tables, and seed them with data.
2. Run the following curl command to authorize your user and get the user id and the a token. Note them both so you can use them in curl commands given for this process. Throughout this process assume that the number in the position of :id in /users/:id for the beginning of any given route is using this id, and [JWT TOKEN HERE] should be replaced with the token returned.
```cURL
curl -X POST \
  http://localhost:3000/auth/login \
  -H 'cache-control: no-cache' \
  -H 'content-type: application/x-www-form-urlencoded' \
  -H 'postman-token: 587da4a3-36fb-f0c9-1948-46f8d70853c3' \
  -d 'email=user%40foobar.com&password=password'
```
3. Run the following CURL command with the appropriate user id and token added.
```cURL
curl -X POST \
  http://localhost:3000/users/2/campaigns/1/calls/start \
  -H 'authorization: JWT [JWT TOKEN HERE]' \
  -H 'cache-control: no-cache' \
  -H 'postman-token: 296588e2-f21f-83f3-7292-d74a0ef05d8c'
```
4. Observe a response with Status 200 and the following XML:
```XML
<?xml version="1.0" encoding="UTF-8"?>
<Response>
    <Say>Welcome, Jeremy. You will be calling on behalf of Campaign 1. Please follow the instructions on your computer screen to start calling.</Say>
    <Play loop="0">https://api.twilio.com/cowbell.mp3</Play>
</Response>
```
The name and campaign name will vary based on what was generated when you seeded the database. Use `npm run db` to connect directly to your database and compare if further verification is needed.
